#************************************************************************************************************
# Patch de scratch-blocks pour ajouter les blocs "ajouter à dire" et "ajouter à penser"
# Ces blocs permettent d'être davantage accessibles que le bloc "regrouper"
# La programmation de ces blocs est dans scratch-vm
#************************************************************************************************************


diff --git a/node_modules/scratch-blocks/blocks_vertical/looks.js b/node_modules/scratch-blocks/blocks_vertical/looks.js
index 66482f0..c83cba4 100644
--- a/node_modules/scratch-blocks/blocks_vertical/looks.js
+++ b/node_modules/scratch-blocks/blocks_vertical/looks.js
@@ -72,6 +72,26 @@ Blockly.Blocks['looks_say'] = {
   }
 };

+Blockly.Blocks['looks_say_more'] = {
+  /**
+   * Block to say more.
+   * @this Blockly.Block
+   */
+  init: function() {
+    this.jsonInit({
+      "message0": Blockly.Msg.LOOKS_SAY_MORE,
+      "args0": [
+        {
+          "type": "input_value",
+          "name": "MESSAGE"
+        }
+      ],
+      "category": Blockly.Categories.looks,
+      "extensions": ["colours_looks", "shape_statement"]
+    });
+  }
+};
+
 Blockly.Blocks['looks_thinkforsecs'] = {
   /**
    * Block to think for some time.
@@ -116,6 +136,26 @@ Blockly.Blocks['looks_think'] = {
   }
 };
 
+Blockly.Blocks['looks_think_more'] = {
+  /**
+   * Block to think more.
+   * @this Blockly.Block
+   */
+  init: function() {
+    this.jsonInit({
+      "message0": Blockly.Msg.LOOKS_THINK_MORE,
+      "args0": [
+        {
+          "type": "input_value",
+          "name": "MESSAGE"
+        }
+      ],
+      "category": Blockly.Categories.looks,
+      "extensions": ["colours_looks", "shape_statement"]
+    });
+  }
+};
+
 Blockly.Blocks['looks_show'] = {
   /**
    * Show block.
diff --git a/node_modules/scratch-blocks/msg/messages.js b/node_modules/scratch-blocks/msg/messages.js
index c7c1e61..8965090 100644
--- a/node_modules/scratch-blocks/msg/messages.js
+++ b/node_modules/scratch-blocks/msg/messages.js
@@ -99,9 +99,11 @@ Blockly.Msg.EVENT_WHENKEYPRESSED_ANY = 'any';
 // Looks blocks
 Blockly.Msg.LOOKS_SAYFORSECS = 'say %1 for %2 seconds';
 Blockly.Msg.LOOKS_SAY = 'say %1';
+Blockly.Msg.LOOKS_SAY_MORE = 'say more %1';
 Blockly.Msg.LOOKS_HELLO = 'Hello!';
 Blockly.Msg.LOOKS_THINKFORSECS = 'think %1 for %2 seconds';
 Blockly.Msg.LOOKS_THINK = 'think %1';
+Blockly.Msg.LOOKS_THINK_MORE = 'think more %1';
 Blockly.Msg.LOOKS_HMM = 'Hmm...';
 Blockly.Msg.LOOKS_SHOW = 'show';
 Blockly.Msg.LOOKS_HIDE = 'hide';
diff --git a/node_modules/scratch-blocks/msg/scratch_msgs.js b/node_modules/scratch-blocks/msg/scratch_msgs.js
index 0a520a4..2694c49 100644
--- a/node_modules/scratch-blocks/msg/scratch_msgs.js
+++ b/node_modules/scratch-blocks/msg/scratch_msgs.js
@@ -7243,9 +7243,11 @@ Blockly.ScratchMsgs.locales["fr"] =
     "EVENT_WHENKEYPRESSED_ANY": "n'importe laquelle",
     "LOOKS_SAYFORSECS": "dire %1 pendant %2 secondes",
     "LOOKS_SAY": "dire %1",
+    "LOOKS_SAY_MORE": "ajouter à dire %1",
     "LOOKS_HELLO": "Bonjour !",
     "LOOKS_THINKFORSECS": "penser à %1 pendant %2 secondes",
     "LOOKS_THINK": "penser à %1",
+    "LOOKS_THINK_MORE": "ajouter à penser %1",
     "LOOKS_HMM": "Hmm...",
     "LOOKS_SHOW": "montrer",
     "LOOKS_HIDE": "cacher",
